logserver/src/main/java/alluxio/logserver/AlluxioLogServerProcess.java

```diff
@@ -0,0 +1,246 @@
+                  remainTimeInMillis -= sleepTimeInMillis;
+                } else {
+                  client.close();
+                  LOG.warn("Connection has been rejected by ExecutorService "
+                      + retryCount + " times till timedout, reason: " + t);
+                  break;
+                }
+              } catch (InterruptedException e) {
+                LOG.warn("Interrupted while waiting to place client on executor queue.");
+                Thread.currentThread().interrupt();
+                break;
+              }
+            } else if (t instanceof Error) {
+              LOG.error("ExecutorService threw error: " + t, t);
+              throw (Error) t;
+            } else {
+              LOG.warn("ExecutorService threw error: " + t, t);
+              break;
+            }
+            // Could not create a thread to serve a client, ignore this client.
+            LOG.warn("Failed to connect with client: {}.", inetAddress.toString());
+            continue;
+          }
+        }
+      } catch (IOException e) {
+        if (!mStopped) {
+          ++failureCount;
+          LOG.warn("Socket transport error occurred during accepting message.", e);
+        }
+        break;
+      }
+    }
+  }
+
+  /**
+   * Called from another thread (actually in the shutdown hook of {@link AlluxioLogServer}
+   * to stop the main thread of {@link AlluxioLogServerProcess}.
+   */
+  private void stopServing() {
+    mStopped = true;
+    if (mServerSocket != null) {
+      try {
+        mServerSocket.close();
+      } catch (IOException e) {
+        LOG.warn("Exception in closing server socket.", e);
+      }
+    }
+
+    mThreadPool.shutdown();
+    long timeoutMS = STOP_TIMEOUT_IN_MS;
+    long now = System.currentTimeMillis();
+    while (timeoutMS >= 0) {
+      try {
+        mThreadPool.awaitTermination(timeoutMS, TimeUnit.MILLISECONDS);
+        break;
+      } catch (InterruptedException e) {
+        long newnow = System.currentTimeMillis();
+        timeoutMS -= (newnow - now);
+        now = newnow;
+      }
+    }
+  }
+
+  /**
+   * Configure a {@link Hierarchy} instance used to retrive logger by name and maintain the logger
+   * hierarchy. An instance of this class will be passed to a {@link AlluxioLog4jSocketNode} so
+   * that the {@link AlluxioLog4jSocketNode} instance can retrieve the logger to log incoming
+   * {@link org.apache.log4j.spi.LoggingEvent}s.
+   *
+   * @param inetAddress inet address of the client
+   * @param logAppenderName name of the appender to use for this client
+   * @return A {@link Hierarchy} instance to pass to {@link AlluxioLog4jSocketNode}
+   * @throws IOException if fails to create an {@link FileInputStream} to read log4j.properties
+   * @throws URISyntaxException if fails to derive a valid URI from the value of property named
+   *         "log4j.configuration"
+   */
+  protected LoggerRepository configureHierarchy(InetAddress inetAddress, String logAppenderName)
```

Comment:
The only caller of this is `AlluxioLog4jSocketNode.java`. Is there a reason why it is in this class?

Reply:
`AlluxioLogServerProcess` is the server process that handles logging clients. `AlluxioLog4jSocketNode` represents one of the many threads forked by `AlluxioLogServerProcess` to service a client. I feel `configureHierarchy` method should be defined in `AlluxioLogServerProcess`. `AlluxioLogServerProcess` passes itself as a "helper" to `AlluxioLog4jSocketNode` instances. When needed, `AlluxioLog4jSocketNode` can invoke the methods of `AlluxioLogServerProcess` to perform certain operations.